### PR TITLE
feat: Zenoti Integration

### PR DESCRIPTION
**Pull request by AfshanKhan**
Documentation: https://docs.erpnext.com/docs/v13/user/manual/en/erpnext_integration/zenoti_integration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ After installation follow user documentation for each integration to set it up.
 
 - Shopify - [User documentation](https://docs.erpnext.com/docs/v13/user/manual/en/erpnext_integration/shopify_integration)
 
+- Zenoti
```
Comment (AfshanKhan) [inline README.md:33]:
```suggestion
- Zenoti - [User documentation](https://docs.erpnext.com/docs/v13/user/manual/en/erpnext_integration/zenoti_integration)
```

File: ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+			list_of_centers = get_list_of_centers()
+			if len(list_of_centers):
+				process_sales_invoices(list_of_centers, error_logs)
+				frappe.db.set_value("Zenoti Settings", "Zenoti Settings", "last_sync", get_datetime())
+				if len(error_logs):
+					make_error_log(error_logs)
+
+
+def sync_stocks():
+	if frappe.db.get_single_value("Zenoti Settings", "enable_zenoti"):
```
Comment (ankush) [inline ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:82]:
```suggestion
	if cint(frappe.db.get_single_value("Zenoti Settings", "enable_zenoti")):
```

Cast this to integer, in some old versions of frappe this casting was improper which caused issues. Best not to rely on that and explicitly cast this.

File: ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+		frappe.throw(
+			_(
+				'Please reconcile the stocks using Stock Reconciliation with purpose "Opening'
+				' Stock" before configuring this'
+			)
+		)
+
+
+def sync_invoices():
+	if frappe.db.get_single_value("Zenoti Settings", "enable_zenoti"):
```
Comment (ankush) [inline ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:67]:
```suggestion
	if cint(frappe.db.get_single_value("Zenoti Settings", "enable_zenoti")):
```

File: ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+				insert_after="zenoti_bussiness_unit_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+		"Sales Invoice": [
+			dict(
+				fieldname="zenoti_invoice_no",
+				label="Zenoti Invoice No",
+				fieldtype="Data",
```
Comment (ankush) [inline ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:215]:
```suggestion
				fieldtype="Data",
```

Sales Invoice doctype is hitting row limits when many integrations are active... is it possible to use `Small Text` for this or `receipt no` field down below?


I've changed this in other integrations (shopify/unicommerce)

File: ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+				label="Zenoti Order No",
+				fieldtype="Data",
+				insert_after="zenoti_order_id",
+				read_only=1,
+				print_hide=1,
+			),
+		],
+	}
+
+	create_custom_fields(custom_fields, update=False)
```
Comment (ankush) [inline ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:332]: `update=False` Intentional?

**Review by ankush — CHANGES_REQUESTED** [review]

File: ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py (hunk 5 above)
Comment (AfshanKhan) [reply ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:332]: Yes I don't want the fields to be updated every time this document is saved.

Comment (ankush) [reply ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py:332]: Hmm. I should also do this and make that flag configurable so we can pass `True` from patches when custom fields do change. 👍

**Review by ankush — APPROVED** [review]

**Pull request merged**